### PR TITLE
dev-java/hamcrest: new package, add 2.2

### PR DESCRIPTION
**Pull request by vaukai**
Signed-off-by: Volkmar W. Pogatzki <gentoo@pogatzki.net>

### CODE REVIEW

File: dev-java/hamcrest/files/hamcrest-2.2-java-11.patch [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ 
+     /**
+#	--- a/hamcrest/src/main/java/org/hamcrest/core/AllOf.java
+#	+++ b/hamcrest/src/main/java/org/hamcrest/core/AllOf.java
+#	@@ -68,7 +68,7 @@ public class AllOf<T> extends DiagnosingMatcher<T> {
+#	      */
+#	     @SafeVarargs
+#	     public static <T> Matcher<T> allOf(Matcher<? super T>... matchers) {
+#	-        return allOf(Arrays.asList(matchers));
+#	+        return allOf((List<Matcher<? super T>> Arrays.asList(matchers));
```
[the remark this answers is not in the document]
Comment (vaukai) [reply dev-java/hamcrest/files/hamcrest-2.2-java-11.patch]: @fordfrog 
Replaced "List" with "Iterable" and now the error messages is [as shown here](https://github.com/gentoo/gentoo/pull/26078/files#diff-716f5c311f82a39286c78c6a5778c86834d7feb91af5d914d9cbc69499230ff9R33-R67).

[the remark this answers is not in the document]
Comment (fordfrog) [reply dev-java/hamcrest/files/hamcrest-2.2-java-11.patch]: probably missing `import java.lang.Iterable;` at the imports at the top of both classes.

[the remark this answers is not in the document]
Comment (vaukai) [reply dev-java/hamcrest/files/hamcrest-2.2-java-11.patch]: > probably missing `import java.lang.Iterable;` at the imports at the top of both classes.

@fordfrog 
Adding that line unfortunately seems not to change anything.  [New output is here](https://github.com/gentoo/gentoo/pull/26078/files#diff-716f5c311f82a39286c78c6a5778c86834d7feb91af5d914d9cbc69499230ff9R31-R67).

File: dev-java/hamcrest/hamcrest-2.2.ebuild [hunk 2]
```diff
@@ -28,9 +28,9 @@ RDEPEND="
 	>=virtual/jre-1.8:*
 "
 
-# With jdk-11 or -17 there are compilation errors.
-# Some of them could be patched but for these 4 errors could not find a solution.
-# hamcrest/src/main/java/org/hamcrest/core/AnyOf.java:49: error: cannot find symbol
+# After adding "import java.lang.Iterable;"
+
+# hamcrest/src/main/java/org/hamcrest/core/AnyOf.java:50: error: cannot find symbol
 #         return anyOf((Iterable<Matcher<? super T>>) Arrays.asIterable(matchers));
```
Comment (fordfrog) [inline dev-java/hamcrest/hamcrest-2.2.ebuild:34]: i searched java doc for `Arrays.asIterable()` but couldn't find it anywhere... the fix might be just to change it to `Arrays.asList()`.

Comment (vaukai) [reply dev-java/hamcrest/hamcrest-2.2.ebuild]: > i searched java doc for `Arrays.asIterable()` but couldn't find it anywhere... the fix might be just to change it to `Arrays.asList()`.

@fordfrog That clearly solves 2 of 4 errors.  It remains 2 errors with [CombinableMatcher.java](https://github.com/gentoo/gentoo/pull/26078/files#diff-716f5c311f82a39286c78c6a5778c86834d7feb91af5d914d9cbc69499230ff9R31-R53)

Comment (fordfrog) [reply dev-java/hamcrest/hamcrest-2.2.ebuild]: here's the updated patch: [hamcrest-2.2-java-11.patch](https://github.com/gentoo/gentoo/files/9034205/hamcrest-2.2-java-11.patch.txt)
with the updated patch it now compiles for me even with `openjdk-17`.

**Pull request closed without merging**